extension/src/validator/validator-builder.js

```diff
@@ -21,7 +21,20 @@ function withPayment (paymentObject) {
           = errorMessages.SUBMIT_ADDITIONAL_PAYMENT_DETAILS_REQUEST_INVALID_JSON
       return this
     },
+    validateReference () {
+      if (!paymentObject.custom || errors.makePaymentRequest)
+        return this
+      if (paymentObject.custom.fields.makePaymentRequest && !paymentObject.custom.fields.makePaymentResponse) {
+        const makePaymentRequestObj = JSON.parse(paymentObject.custom.fields.makePaymentRequest)
+        if (!makePaymentRequestObj.reference)
+          errors.missingReference = errorMessages.MAKE_PAYMENT_REQUEST_MISSING_REFERENCE
+      }
+      return this
+    },
     validateAmountPlanned () {
+      if (!paymentObject.custom)
```

Comment:
This one is not related to the PR issue? Maybe you can add a short comment on why this check is necessary?

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/395/commits/b0075e26286a50b5eb6004883a28d8b51bb11b11